File: docs/examples/logical-blocks/recipe.js
import {HtmlBuilder} from 'gramlot-dom';

export class PriceRecipe extends HtmlBuilder {
    main(root) {
        const card = root.div({datapath: 'order'});
        card.dataSetter({destination: '.quantity', value: 2});
        card.dataSetter({destination: '.price', value: 5});
        card.dataFormula({
            destination: '.total', formula: 'quantity * price',
            quantity: '^.quantity', price: '=.price', _on_start: true,
        });
        card.dataController({
            func: 'sourceNode.SET(".summary", `Total: ${total}`)',
            total: '^.total', _on_start: true,
        });
        card.span('==quantity * price', {quantity: '^.quantity', price: '=.price'});
        card.p('^.summary');
    }
}

File: docs/examples/logical-blocks/recipe.py
from gramlot.builder import GramlotBuilder


class PriceRecipe(GramlotBuilder):
    def main(self, root):
        card = root.div(datapath='order')
        card.dataSetter('.quantity', 2)
        card.dataSetter('.price', 5)
        card.dataFormula(
            '.total', 'quantity * price',
            quantity='^.quantity', price='=.price', _on_start=True,
        )
        card.dataController(
            'sourceNode.SET(".summary", `Total: ${total}`)',
            total='^.total', _on_start=True,
        )
        card.span('==quantity * price', quantity='^.quantity', price='=.price')
        card.p('^.summary')
